### PR TITLE
[CGPROD-2766] shop list view

### PR DESCRIPTION
**Pull request by BeeGodwin**
- Removed items from shop demo json5, replaced with a reference to a catalogue key
- Scrollable List takes this catalogue key and uses it to retrieve a list of items
- Item list is used in existing code to parameterise gel button overlay
- added scaling for the item icon, tweaked demo values to be closer to the design
- sync'd the shop config in the theme with the (still evolving) config in debug
- added sample assets & text

### CODE REVIEW

File: src/core/layout/scrollable-list/scrollable-list.js [hunk 1]
```diff
@@ -39,21 +41,22 @@ const getPanelY = scene => {
...
-const createInnerPanel = scene => {
+const createInnerPanel = (scene, collection) => {
     const sizer = scene.rexUI.add.sizer({ orientation: "x", space: { item: 0 } });
-    sizer.add(createTable(scene), { expand: true });
+    sizer.add(createTable(scene, collection), { expand: true });
     return sizer;
 };
 
-const createTable = scene => {
+const createTable = (scene, collection) => {
```
Comment (BeeGodwin) [inline src/core/layout/scrollable-list/scrollable-list.js:50]: once done, will get the collection here once passing it down.

File: src/core/layout/scrollable-list/scrollable-list.js [hunk 2]
```diff
@@ -101,6 +105,7 @@ const getPanelItems = panel => panel.getByName("grid", true).getElement("items")
 export class ScrollableList extends Phaser.GameObjects.Container {
     constructor(scene) {
         super(scene, 0, 0);
+        // this.collectionKey = "armoury";
```
Comment (BeeGodwin) [inline src/core/layout/scrollable-list/scrollable-list.js:108]:
```suggestion
```

File: src/core/layout/scrollable-list/scrollable-list.js [hunk 3]
```diff
@@ -46,14 +47,17 @@ const createInnerPanel = scene => {
...
     const table = scene.rexUI.add.gridSizer({
         column: 1,
-        row: scene.config.items.length,
+        row: collection.length,
         space: { row: scene.config.listPadding.y },
         name: "grid",
     });
 
-    scene.config.items.forEach((item, idx) => {
+    collection.forEach((item, idx) => {
```
Comment (FostUK) [inline src/core/layout/scrollable-list/scrollable-list.js:60]: Can be one liner

**Review by FostUK — APPROVED** [review]
Looks great!
🦙

**Pull request merged**